src/libs/bir_expLib.sml

```diff
@@ -156,8 +137,15 @@ struct
           ((xf "(") cf (ef expm) cf (xf ":") cf (xf endistr) cf (xf "[") cf (ef expad) cf (xf "] = ") cf (ef expv) cf (xf ")"))
         end
 
+      else if is_bir_exp_imp exp then
```

Comment:
this could be a problem because this whole function assumes "normalized" terms which only contain the basic BIR constructors from core

Reply:
I didn't see that in the documentation :stuck_out_tongue: Sure we could remove that, but then we should provide a function to "normalize" a BIR expression. But I don't really see the point of restricting it :slightly_smiling_face:

Reply:
the problem is that this kind of function always has some scope and i find the basic set of expression types a natural fit unless we have another one. but i would leave it like this for now also. it needs to be revisited anyways I think. and possibly we have other infrastructure around at that time to describe a clear scope or similar.